Don't treat DW_AT_const_value as meaning "static" in Swift.

scallanan opened the pull request:
This is a heuristic that works for code compiled with Clang, but the
Swift language is different (as explained in an added comment).

I've also added a test that checks this.

rdar://31332056

scallanan committed:
Don't treat DW_AT_const_value as meaning "static" in Swift.

This is a heuristic that works for code compiled with Clang, but the
Swift language is different (as explained in an added comment).

I've also added a test that checks this.

rdar://31332056

scallanan commented:
@swift-ci please test

jasonmolenda commented on source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp:
```diff
@@ -4268,7 +4268,15 @@ VariableSP SymbolFileDWARF::ParseVariableDIE(const SymbolContext &sc,
         } else {
-          if (location_is_const_value_data)
+          // The heuristic for inferring static variables works for Clang's
+          // behavior on C-like languages, which generallydoes n ot emit
```
your keyboard had some sync issues with the space bar.

jimingham commented:
What happens with module-level const variables?  In:

```
let g_int : Int = 10

func use_g_int()
{
    let local_int : Int = 20
    print(g_int, local_int)
}

use_g_int()

```

When stopped at the print above, does g_int show up in "frame var"?

scallanan commented:
I will add that to the test, thanks Jim.  I will also fix the misplaced space in the comment, thanks Jason!

scallanan committed:
Merged origin/stable

scallanan committed:
Fixed a typo in SymbolFileDWARF

scallanan committed:
Ensure that global lets don't make it into the locals list

scallanan commented:
Updated.  The code actually is only distinguishing between locals and statics; globals follow a different path which this doesn't touch at all.  I've added a global to the test and made sure it doesn't sneak its way into the list, though, so that this assumption is documented.

jimingham reviewed (approved):
Great.  Then this looks good to me.
